contracts/UBI.sol

```diff
@@ -57,8 +57,15 @@ contract UBI is Initializable {
 
   mapping (address => mapping (address => uint256)) public allowance;
 
-  /// @dev A lower bound of the total supply. Does not take into account tokens minted as UBI by an address before it moves those (transfer or burn).
-  uint256 public totalSupply;
+  /**@dev M0 supply marker
+  * M0 should be updated after.. 
+  * a) adding new PoH
+  * b) revoking active PoH
+  *
+  * If accrued per sec rate is changed,
+  * It'll fluctuate +- "real" total supply
+  */
+  uint256 public supplyMarker; // same slot/position as old totalSupply to match upgradable pattern
```

Comment:
I like this change to calculate a more accurate total supply, consider making it a separate pull request since the streaming feature doesn't depend on it. I have a feeling if the PR is smaller it will make it in faster.